### PR TITLE
light: expand on errors and docs

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Updates light docs with a section on security and adds a more informative error in the case of a light client attack.

### CODE REVIEW

File: docs/tendermint-core/light-client.md [hunk 1]
```diff
@@ -30,11 +31,26 @@ known validator set. From there, all the application state is verifiable with
...
+is correct. As Tendermint consensus can withstand 1/3 byzantine faults, this is
+the default trust level, however, for greater security you can increase it (max:  
+1).
+
+Similar to a full node, light clients can also be subject to byzantine attacks.
+A light client also runs a detector process which cross verifies headers from a
+primary with witnesses. Therefore light clients should be set with enough witnesses.
 
-[Trust Options](https://pkg.go.dev/github.com/tendermint/tendermint/light?tab=doc#TrustOptions)
+If the detector observes a faulty provider it will report it to another provider
```
Comment (melekes) [inline docs/tendermint-core/light-client.md:48]: I think we should state that the light client is not **safe** when a) more than {trust_level} of validators are malicious b) all witnesses are malicious

Comment (cmwaters) [reply docs/tendermint-core/light-client.md:48]: Sure, this would be in replacement of lines 44 - 49 right? or the entire section

Comment (melekes) [reply docs/tendermint-core/light-client.md:48]: not a replacement, but rather an addition which clearly states the fact.

File: light/errors.go [hunk 2]
```diff
@@ -65,6 +65,12 @@ func (e ErrVerificationFailed) Error() string {
 		e.From, e.To, e.Reason)
 }
 
+// ErrLightClientAttack is returned when the light client has detected an attempt
+// to verify a false header and has sent the evidence to either a witness or primary.
+var ErrLightClientAttack = errors.New("ATTACK DETECTED. Light client received valid conflicting header from witness." +
```
Comment (tessr) [inline light/errors.go:70]:
```suggestion
var ErrLightClientAttack = errors.New("Attempted attack detected. Light client received valid conflicting header from witness." +
```

**Comment by tessr** [comment]
This looks pretty good to me. Maybe we can add a message to the logs for detected attack attempts that's something like 

> The light client has detected, and prevented, an attempted amnesia attack.  We think this attack is pretty unlikely, so if you see it, that's interesting to us. Can you let us know by opening an issue through https://github.com/tendermint/tendermint/issues/new?

**Review by tessr — APPROVED** [review]

**Pull request merged**